src/client/Microsoft.Identity.Client/TelemetryCore/TelemetryManager.cs

```diff
@@ -64,6 +70,9 @@ public void StartEvent(EventBase eventToStart)
 
         public void StopEvent(EventBase eventToStop)
         {
+            ProcessEventsForCurrentHttpTelemetryContent(eventToStop);
```

Comment:
Does this need locking?  We could have more than one request going on at the same time which means we could be calling StopEvent in the telemetry manager more than once.  and this call pattern isn't idempotent or atomic.